File: my_cpp_pkg/src/activity2_first_node.cpp
#include <rclcpp/rclcpp.hpp>
#include <example_interfaces/msg/int64.hpp>

class FirstNode : public rclcpp::Node
{
public:
    FirstNode() : Node("first_node")
    {
        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&FirstNode::publish_number, this)); 

        RCLCPP_INFO(this->get_logger(), "First Node has been started!");
    }

private:
    int64_t num = 2;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

    void publish_number()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = num;
        publisher_->publish(msg);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<FirstNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
}

File: my_py_pkg/my_py_pkg/activity2_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import Int64

class FirstNode(Node):
    def __init__(self):
        super().__init__("first_node")

        self.topic_name = "number"
        self.num = 2

        self.publisher_ = self.create_publisher(Int64, self.topic_name, 10)
        self.timer_ = self.create_timer(1, self.publish_number)
        self.get_logger().info("First Node has been started!")

    def publish_number(self):
        msg = Int64()
        msg.data = self.num
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = FirstNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()